src/components/Stepper.tsx

```diff
@@ -22,13 +22,8 @@ export function Stepper({ steps, currentStep, onChange }: StepperBarProps) {
     throw new Error("Stepper must be initialized with at least one step");
   }
 
-  const [progressBarStep, setProgressBarStep] = useState(0);
-  useEffect(() => {
-    const currentStepIndex = steps.findIndex((s: Step) => s.value === currentStep) || 0;
-    if (currentStepIndex > progressBarStep) {
-      setProgressBarStep(currentStepIndex);
-    }
-  }, [currentStep]);
+  // calc progress based on last completed step - return -1 when no steps completed
+  const progressBarStep = steps.reduce((acc, step, idx) => (step.state === "complete" ? idx : acc), -1);
```

Comment:
Could we rename this to `lastCompletedStep` then? Will this prevent a user from going back steps?

Reply:
Will defer to Dean, but I believe the idea is that they could go back, but the progress bar locks at the most-completed-so-far step.

Reply:
ahh, okay. Been a while since I looked at this. That makes sense.

Reply:
Happy to rename variable to `lastCompletedStep`. This only impacts the progress bar "visuals" in terms of how much "fill" it has. The user can still navigate to different steps as before.